tests/integration/test_data_cli.py

```diff
@@ -2,19 +2,86 @@
 
 from click.testing import CliRunner
 import pytest
-
+import json
+import httpx
+import respx
+from http import HTTPStatus
 from planet.cli import cli
 
+TEST_URL = 'https://api.planet.com/data/v1'
+TEST_QUICKSEARCH_URL = f'{TEST_URL}/quick-search'
+
+
+@pytest.fixture
+def invoke():
+
+    def _invoke(extra_args, runner=None):
+        runner = runner or CliRunner()
+        args = ['data'] + extra_args
+        return runner.invoke(cli.main, args=args)
 
-def test_data_command_registered():
+    return _invoke
+
+
+def test_data_command_registered(invoke):
     """planet-data command prints help and usage message."""
-    result = CliRunner().invoke(cli.main, ["data", "--help"])
+    runner = CliRunner()
+    result = invoke(["--help"], runner = runner)
     assert result.exit_code == 0
     assert "Usage" in result.output
     assert "search-quick" in result.output
     # Add other sub-commands here.
 
 
+@respx.mock
+@pytest.mark.asyncio
+@pytest.mark.parametrize("filter", [{''}, {"foo"}])
+@pytest.mark.parametrize(
+    "item_types", ['PSScene', 'SkySatScene', ('PSScene', 'SkySatScene')])
+def test_data_search_quick_filter_fail(invoke, item_types, filter):
+    """Test for planet data search_quick. Test with multiple item_types.
+    Test should fail as filter does not contain valid JSON."""
+    mock_resp = httpx.Response(HTTPStatus.OK,
+                               json={'features': [{
+                                   "key": "value"
+                               }]})
+    respx.post(TEST_QUICKSEARCH_URL).return_value = mock_resp
+
+    runner = CliRunner()
+    item_types = 'SkySatScene'
+    with pytest.raises(TypeError):
```

Comment:
@mkshah605 to answer your question: pytest.raises is just as good as an assert statement when checking that code raises an exception.

However, we intend that the CLI does not let exceptions propagate all the way out. In the integration tests we should assert `assert result.exit_code == 1` when we are testing for failure. If a TypeError is being raised, it's likely that there is a bug.

Reply:
@sgillies here's the TypeError I'm getting when I give bad inputs like {"foo"} or {''} to the quick-search function filter argument: `TypeError: Object of type set is not JSON serializable`. When I run the code (as a user) this way: `planet data search-quick PSScene {'foo'}  ` I get what I expect: `Error: File does not contain valid json.`, which I've put in the `parse_filter` function in `planet/cli/data.py`.

Reply:
@mkshah605 I found the problem! I will put another comment inline.